state_version.go

```diff
@@ -243,6 +273,46 @@ func (s *stateVersions) Create(ctx context.Context, workspaceID string, options
 	return sv, nil
 }
 
+func (s *stateVersions) putURL(ctx context.Context, putURL string, data []byte) error {
+	reader := bytes.NewReader(data)
+	req, err := s.client.NewRequest("PUT", putURL, reader)
+	if err != nil {
+		return err
+	}
+
+	return req.Do(ctx, nil)
+}
```

Comment:
This code is duplicated in `UploadTarGzip()` so I think this should belong as a `Client` receiver method:

```go
// or don't export this
func (c *Client) PutArchivistURL(ctx context.Context, archivistURL string, reader io.Reader) error {
  req, err := c.NewRequest("PUT", archivistURL, reader)
  // error handling omitted
  return req.Do(ctx, nil)
}
```

It would eliminate the code duplication with `UploadTarGzip()` in `RegistryModules`, `ConfigurationVersions` and `Policies` 

What do you think?

Reply:
OK. I don't think it should be exported but definitely shared.

Reply:
This turned into an adventure! See my notes in the PR description under "Refactor Notes"